pipeline-model-definition/src/main/groovy/org/jenkinsci/plugins/pipeline/modeldefinition/model/Environment.groovy

```diff
@@ -60,7 +60,11 @@ public class Environment implements Serializable {
 
     public Map<String,String> getCredsMap(CpsScript script) {
         Map<String,String> resolvedMap = new TreeMap<>()
-        EnvVars contextEnv = new EnvVars((Map<String,String>)script.getProperty("params"))
+        EnvVars contextEnv = new EnvVars()
+        ((Map<String,Object>)script.getProperty("params")).each { k, v ->
+            contextEnv.put(k, v.toString())
```

Comment:
Turns out a null parameter causes other problems entirely -
```
java.lang.NullPointerException
	at org.jenkinsci.plugins.workflow.cps.persistence.IteratorHack.writeReplace(IteratorHack.java:52)
	at org.jenkinsci.plugins.workflow.cps.ParamsVariable.addValue(ParamsVariable.java:88)
	at org.jenkinsci.plugins.workflow.cps.ParamsVariable.getValue(ParamsVariable.java:75)
	at org.jenkinsci.plugins.workflow.cps.CpsScript.getProperty(CpsScript.java:121)
```
I'll add some NPE checks but that's a bug in `workflow-cps`, I think.